migrations/versions/0318_update_postage_constraint_1.py

```diff
@@ -0,0 +1,34 @@
+"""
+
+Revision ID: 0318_update_postage_constraint_1
+Revises: 0317_uploads_for_all
+Create Date: 2020-03-11 12:01:41.533192
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = '0318_update_postage_constraint_1'
+down_revision = '0317_uploads_for_all'
+
+
+def upgrade():
+    op.drop_constraint('chk_notifications_postage_null', 'notifications')
+    op.drop_constraint('chk_notification_history_postage_null', 'notification_history')
+
+    op.execute("""
+        ALTER TABLE notifications ADD CONSTRAINT "chk_notifications_postage_null"
+        CHECK (
+            CASE WHEN notification_type = 'letter' THEN
+                postage is not null and postage in ('first', 'second', 'Europe', 'rest of world')
+            ELSE
+                postage is null
+            END
+        )
+        NOT VALID
+    """)
+
+
+def downgrade():
+    pass
```

Comment:
Why no downgrade?

Reply:
Because although the change is reversible, I thought that ideally it shouldn't be downgraded in only 3 migrations since that might lock access to the tables for too long. This is because if we downgrade we'd need to add the constraint back to `notifications_history` too, which means (I think) we'd want 4 migrations to downgrade:

1. drop constraints from `templates` and `templates_history` and add the original constraints with `NOT VALID`
2. drop constraint from `notifications` and add the original constraint with `NOT VALID`
3. drop constraint from `notification_history` and add the original constraint with `NOT VALID`
4. validate all 4 constraints

Not sure if there's a way round this, or if it would be ok to somehow fit all of that into 3 migrations after all?